### ISSUE TITLE
Consider using for_each instead of count

### ISSUE THREAD

**Issue by kerma** [mhmdio/terraform-aws-client-vpn-federated-authentication#1]
Resources like aws_ec2_client_vpn_route.additional are currently defined with counts. Please consider using maps and for_each instead. In that case the order of which the resources are defined does not matter and there are no changes (forces replacement) in the plan when the user is redefining the order.

The main use case for this is a place where those variables are computed or defined dynamically, using data resources or other means. This way the user doesn't even directly control which gets passed to the module and recreating routes when things change may be confusing and unnecessary.

Also, it's a good chance that a race condition happens when routes are changiing. User will see something like this and another apply is required:

```
Error: error creating client VPN route "cvpn-endpoint-094cfee03axxxxxxx,subnet-xxxxxxx33632232e8,10.0.0.0/16":
InvalidClientVpnDuplicateRoute: Route already exists
│ 	status code: 400, request id: 282b5905-3755-4d23-9b3e-074eaxxxxxxx
```

**Comment by mhmdio** [comment]
Yes, good catch I'm welling to update that, or PR is most welcome

**Comment by mhmdio** [comment]
fixed in tag v2.0.0
